src/components/Categories/Categories.component.jsx

```diff
@@ -1,21 +1,33 @@
 import React from "react";
-import categories  from "../../mocks/product-categories.json"
+import { Link } from "react-router-dom";
+import { useCategories } from "../../utils/hooks/useCategories";
 
 function Categories() {
-    const { results } = categories
-    return (
-        <div className="categories">
-            <h2>Categories</h2>
-            <div className="categories-grid">
-                {results.map(category => (
-                    <div className="category" key={category.id}>
-                        <img className="category-image" src={category.data.main_image.url} alt={category.data.main_image.alt} />
-                        <h3>{category.data.name}</h3>
-                    </div>
-                ))}
+  const { data: catList, isLoading: catIsLoading } = useCategories();
+
+  return (
+    <div className="categories">
+      <h2>Categories</h2>
+      {catIsLoading ? (
+        <p>Loading...</p>
+      ) : (
+        <div className="categories-grid">
+          {catList.results.map((category) => (
+            <div className="category" key={category.id}>
```

Comment:
Something you can do to make this component even better is create a component that handles what the category renders. For example a category card or something like that that just receives the data and is used with the only purpose of rendering the category
```suggestion
            {
            catList.results.map((category) => (
              <Card {...category} /> 
              // or Container 
            )
            }
```
or something like that that in the future may be used in other section. Imagine you want to show also the categories on the bottom of a product or in the side, you can just send the information to the card and have a different container that handles the side of your cards

Reply:
Got it, like the single Product component! I'll work on it.